cni/pkg/nodeagent/informers.go

```diff
@@ -212,19 +216,42 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 		wasAnnotated := oldPod.Annotations != nil && oldPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		isAnnotated := newPod.Annotations != nil && newPod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionEnabled
 		shouldBeEnabled := util.PodRedirectionEnabled(ns, newPod)
+		wasTerminated := kube.CheckPodTerminal(oldPod)
+		isTerminated := kube.CheckPodTerminal(newPod)
+		// only checks if the pod state just changed from not-terminated to terminated.
+		// if pod state changed in *any* other way, this should be false.
+		justTerminated := !wasTerminated && isTerminated
+
+		// Check intent (labels) versus status (annotation) - is there a delta we need to fix?
+		changeNeeded := (isAnnotated != shouldBeEnabled) && !isTerminated
+
+		// nolint: lll
+		log.Debugf("pod %s events: wasAnnotated(%v), isAnnotated(%v), shouldBeEnabled(%v), changeNeeded(%v), wasTerminated(%v), isTerminated(%v), oldPod(%+v), newPod(%+v)",
+			pod.Name, wasAnnotated, isAnnotated, shouldBeEnabled, changeNeeded, wasTerminated, isTerminated, oldPod.ObjectMeta, newPod.ObjectMeta)
+
+		// If it was a job pod that (a) we captured and (b) just terminated (successfully or otherwise)
+		// remove it (the pod process is gone, but kube will keep the Pods around in
+		// a terminated || failed state - we should still do cleanup)
+		//
+		// Note that kube may either restart the same pod, or spawn a new one, depending on how
+		// the job/cronjob is configured. Either way, we will come back thru here.
+		if isAnnotated && justTerminated {
+			log.Debugf("deleting pod %s from mesh, reason: isAnnotated(%v), justTerminated(%v)", newPod.Name, isAnnotated, justTerminated)
+			// Unlike the other cases, we actually want to use the "old" event for terminated job pods
+			// - kubernetes will (weirdly) clear the ip from the pod status on termination (boo)
```

Comment:
Really? i am not seeing that

Reply:
That is exactly what I am doing, i never see the IP go away

Reply:
https://gist.githubusercontent.com/bleggett/aaaa42563bc9f4f14edfd39594672f13/raw/d0fa4289c882da0881b645d121e112ded8dcb8b6/gistfile1.txt

It's because we're checking `Pod.Status` for the IP, and as you can see from this debug dump, the most _recent_ status for this initial "pod terminated" event lacks an IP.

We get a subsequent event where it's added back as a new status.

We either stall the removal and wait for some future event that adds it back, or check the old event pod, which will always have the IP we want anyway.

Reply:
Oh what, I see that too now.. missed it before. That is bizarrare, maybe a k8s bug